src/pages/LoginPage.jsx

```diff
@@ -0,0 +1,151 @@
+import CircularProgress from "@mui/material/CircularProgress";
+import React, { useEffect, useRef, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { login } from "../apis/user";
+import AlertBox from "../components/common/accounts/AlertBox";
+import InputGroup from "../components/common/accounts/InputGroup";
+import Box from "../components/common/atoms/Box";
+import Button from "../components/common/atoms/Button";
+import Container from "../components/common/atoms/Container";
+import useInput from "../hooks/useInput";
+import { fetchUserInfo, logIn } from "../store/slices/userSlice";
+
+export default function LoginPage() {
+  const navigate = useNavigate();
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false); // login api 호출 중인지 아닌지 확인
+  const { values, handleChange } = useInput({
+    email: "",
+    password: "",
+  });
+  const emailInputRef = useRef(null);
+  const passwordInputRef = useRef(null);
+  const dispatch = useDispatch();
+
+  const isValidValue = () => {
+    const emailRegex =
+      /([\w-.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/;
+    const passwordRegex =
+      /^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,20}$/;
+
+    return emailRegex.test(values.email) && passwordRegex.test(values.password);
+  };
+
+  const handleLogin = async () => {
+    if (!values.email) {
+      console.log(values.email);
+      setErrorMessage("이메일을 입력해주세요.");
+      emailInputRef.current.focus();
+      return;
+    }
+    if (!values.password) {
+      setErrorMessage("비밀번호를 입력해주세요.");
+      passwordInputRef.current.focus();
+      return;
+    }
+    if (!isValidValue()) {
+      // validation check fail시 api 호출하지 않음
+      setErrorMessage("");
+      setIsSubmitting(true);
+      await new Promise((resolve) => {
+        setTimeout(resolve, 500);
+      });
+      setErrorMessage("이메일 또는 비밀번호를 잘못 입력했습니다. ");
+      setIsSubmitting(false);
```

Comment:
유효성 검사 실패시 0.5초 로딩하는 것처럼 보여주다가 에러를 표시하려고 이렇게 작성해주신 것 같아요
async-await로 500ms 지연시켰는데 지연된 후에 작성된 로직을 setTimeout 안에 넣어버려도 좋을 것 같아요!
```suggestion
      setTimeout(() => {
        setErrorMessage("이메일 또는 비밀번호를 잘못 입력했습니다. ");
        setIsSubmitting(false);
      }, 500);
```

Reply:
수정하였습니다!